pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,228 @@
+package localclusterimport
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/agent"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const LocalClusterNamespace = "local-cluster"
+
+type LocalClusterImport struct {
+	clusterImportOperations ClusterImportOperations
+	log                     *logrus.Logger
+}
+
+func NewLocalClusterImport(localClusterImportOperations ClusterImportOperations, log *logrus.Logger) LocalClusterImport {
+	return LocalClusterImport{clusterImportOperations: localClusterImportOperations, log: log}
+}
+
+func (i *LocalClusterImport) shouldImportLocalCluster() (bool, error) {
+	// The presence of agentClusterInstall for the local cluster will determine whether or not we need to perform an import
+	agentClusterInstall, err := i.clusterImportOperations.GetAgentClusterInstall(LocalClusterNamespace, LocalClusterNamespace+"-cluster-install")
+	if err != nil && !k8serrors.IsNotFound(err) {
+		i.log.Errorf("hubClusterDay2Importer: There was an error fetching the AgentClusterInstall while checking for need to register local cluster.")
+		return false, err
+	}
+	// TODO: We need to determine what to do if there has been a recent upgrade
+	// See MGMT-15705 for more information.
+	if agentClusterInstall != nil {
+		i.log.Infof("hubClusterDay2Importer: Found AgentClusterInstall for hub cluster, assuming that hub has been correctly registered for ZTP day 2 operations: %s", agentClusterInstall.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (i *LocalClusterImport) ImportLocalCluster() error {
```

Comment:
Few things about this function:
1. It does to many things, which should be split to sub functions for both readability and tests.
2. It can potentially fail in too many places (which is related to the first point, in part). I also think that for many things here it is worth to use multiErr to aggregate a much of the problems and reflect them to the user as early as possible.

Reply:
Adding a feature flag to options

Reply:
1: Done
2: Done